lib/hyrax/transactions/steps/save.rb

```diff
@@ -37,6 +37,7 @@ def call(change_set, user: nil)
           saved.permission_manager.acl.permissions = unsaved.permission_manager.acl.permissions if
             unsaved.respond_to?(:permission_manager)
 
+          user ||= ::User.find_by_user_key(saved.depositor)
```

Comment:
it's increasingly important that we report some user as having been responsible for published events. this infers that it was the `depositor` unless we provided a user explicitly with `step_args`.